website/blog/unusual-builds-w-bytes.md

```diff
@@ -0,0 +1,224 @@
+In case of input downloads, the target ID would be the consuming/child action's label while in the case of output downloads, the producing/parent action label would be used.
+
+Once we had [upgraded to a Bazel version](./blog/bisect-bazel) with this change, we were able to see that the unwanted downloads were coming from the `output` downloads.
+
+## The Bug
+
+We discovered another code path in Bazel that can trigger unexpected downloads: when a blob's TTL expires.
+
+> So... What is a blob's TTL? 🤔
+
+As an artifact-oriented build system, Bazel relies heavily on artifacts existing in its output path to determine if a build is up-to-date.
+If an artifact exists and matches the expected result, then Bazel will skip the action that produces the artifact.
+However, in a remote build with minimal download, Bazel does not have the artifact locally to validate against and must establish some trust that the artifact was created and stored in the Remote Cache.
+
+Introduced in [PR#17639](https://github.com/bazelbuild/bazel/pull/17639), this "trust" is added to Bazel by storing the expected Time To Live (TTL) of the artifact in the Remote Cache.
+When Bazel executes an action remotely, instead of downloading the outputs referenced inside the ActionResult, it instead just stores the references inside its in-process analysis cache (aka. Skyframe) with an expected TTL that is determined by the flag `--experimental_remote_cache_ttl` (default: 3 hours).
+When the TTL expires, instead of checking if the output artifacts are up-to-date remotely, Bazel downloads the entire artifact to disk.
+We were able to validate this by setting the flag `--experimental_remote_cache_ttl=0s` to eagerly trigger the downloads.
+
+```bash
+# Initial build
+$ bazel clean && bazel test //cli/... --config=remote-minimal --disk_cache= --experimental_remote_cache_ttl=1d
+$ du -h $(bazel info output_base)/execroot | sort -h
+...
+1.8M	/private/var/tmp/_bazel_fmeum/412888b82b4f18156bc415025cb8faa1/execroot
+
+# Incremental build
+$ bazel clean && bazel test //cli/... --config=remote-minimal --disk_cache= --experimental_remote_cache_ttl=1d
+$ du -h $(bazel info output_base)/execroot | sort -h
+...
+1.8M	/private/var/tmp/_bazel_fmeum/412888b82b4f18156bc415025cb8faa1/execroot
+
+# Incremental build with TTL expired
+$ bazel clean && bazel test //cli/... --config=remote-minimal --disk_cache= --experimental_remote_cache_ttl=0
+$ du -h $(bazel info output_base)/execroot | sort -h
+...
+640M	/private/var/tmp/_bazel_fmeum/412888b82b4f18156bc415025cb8faa1/execroot
+```
+
+We fixed this issue in [PR#25398](https://github.com/bazelbuild/bazel/pull/25398) by enhacing the logic that validates Bazel's blob metadata and reinforcing it with additional tests.
+
+Worth noting that there was also a draft fix 8 months ago by a Bazel community member, [David Sanderson](https://github.com/dws), in [PR#23066](https://github.com/bazelbuild/bazel/pull/23066).
```

Comment:
It would make more sense to move this down to the workaround section (perhaps after the smiley quote) and making clear that the draft PR is still relevant as follows:

```suggestion
Worth noting that there is also a [draft PR](https://github.com/bazelbuild/bazel/pull/23066) by Bazel community member [David Sanderson](https://github.com/dws), which allows to bind the TTL to the lifetime of the Bazel server. This could have provided a more robust workaround by avoiding the need to hardcode a long TTL.
```